src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,581 @@
+
+    params.setValue("min_score", 0.7, "The minimum score a spectrum must have to be assignable to a transition.");
+    params.setMinFloat("min_score", 0.0);
+
+    params.setValue("min_forward_match", 0.9, "Minimum forward match.");
+    params.setMinFloat("min_forward_match", 0.0);
+    params.setMaxFloat("min_forward_match", 1.0);
+
+    params.setValue("min_reverse_match", 0.9, "Minimum reverse match.");
+    params.setMinFloat("min_reverse_match", 0.0);
+    params.setMaxFloat("min_reverse_match", 1.0);
+
+    params.setValue("mz_tolerance", 0.1, "Mass to Charge tolerance.");
+
+    params.setValue("mz_unit_is_Da", "true", "True if the Mass to Charge tolerance unit is Da. False if ppm.");
+    params.setValidStrings("mz_unit_is_Da", ListUtils::create<String>("false,true"));
+
+    params.setValue("sgolay_frame_length", 15, "The number of subsequent data points used for smoothing.\nThis number has to be uneven. If it is not, 1 will be added.");
+    params.setValue("sgolay_polynomial_order", 3, "Order of the polynomial that is fitted.");
+    params.setValue("gauss_width", 0.2, "Gaussian width in Da or ppm, estimated peak size.");
+    params.setValue("use_gauss", "true", "Use Gaussian filter for smoothing (alternative is Savitzky-Golay filter)");
+    params.setValidStrings("use_gauss", ListUtils::create<String>("false,true"));
+    params.setValue("signal_to_noise", 1.0, "Signal-to-noise threshold at which a peak will not be extended any more. Note that setting this too high (e.g. 1.0) can lead to peaks whose flanks are not fully captured.");
+    params.setMinFloat("signal_to_noise", 0.0);
+
+    params.setValue("peak_height_min", 0.0, "Minimum picked peak's height.");
+    params.setMinFloat("peak_height_min", 0.0);
+    params.setValue("peak_height_max", 4e6, "Maximum picked peak's height.");
+    params.setMinFloat("peak_height_max", 0.0);
+    params.setValue("fwhm_threshold", 0.0, "Picked peak's FWHM threshold.");
+    params.setMinFloat("fwhm_threshold", 0.0);
+
+    params.setValue("tic_weight", 1.0, "TIC weight when scoring spectra.");
+    params.setMinFloat("tic_weight", 0.0);
+    params.setValue("fwhm_weight", 1.0, "FWHM weight when scoring spectra.");
+    params.setMinFloat("fwhm_weight", 0.0);
+    params.setValue("snr_weight", 1.0, "SNR weight when scoring spectra.");
+    params.setMinFloat("snr_weight", 0.0);
+  }
+
+  void TargetedSpectraExtractor::annotateSpectra(
+    const std::vector<MSSpectrum>& spectra,
+    const TargetedExperiment& targeted_exp,
+    std::vector<MSSpectrum>& annotated_spectra,
+    FeatureMap& features
+  )
+  {
+    annotated_spectra.clear();
+    features.clear(true);
+    const std::vector<ReactionMonitoringTransition> transitions = targeted_exp.getTransitions();
+    for (UInt i=0; i<spectra.size(); ++i)
+    {
+      MSSpectrum spectrum = spectra[i];
+      // It is supposed to have RT in minutes in target list file, therefore we divide by 60.0
```

Comment:
hmmm ... please note that the user may use seconds or minutes in the TraML file, so I would be careful about that assumption

Reply:
especially as the default assumption of the CSV to TraML converter assumes seconds